addons/web_editor/tools.py

```diff
@@ -0,0 +1,133 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+
+from markupsafe import Markup
+
+from odoo import _
+from odoo.tools import image_process
+from odoo.modules.module import get_module_resource
+
+
+# To detect if we have a valid URL or not
+valid_url_regex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+# Regex for few of the widely used video hosting services
+player_regexes = {
+    'youtube': r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})\S*$',
+    'vimeo': r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*',
+    'dailymotion': r'(?:.+dailymotion.com\/(video|hub|embed\/video|embed)|dai\.ly)\/([^_?]+)[^#]*(#video=([^_&]+))?',
+    'instagram': r'(?:(.*)instagram.com|instagr\.am)\/p\/(.[a-zA-Z0-9-_\.]*)',
+    'youku': r'(.*).youku\.com\/(v_show\/id_|embed\/)(.+)',
```

Comment:
The dailymotion and youku ID matches seem very permissive given we're just shoving them straight through the output.

Reply:
I'll check for a fix in stable for this